Missing SV frequencies?

I just had a chat with two users who had noticed an abundance of seemingly rare SVs on their cases, or with other words a paucity of  variants reporting as local. Is loqusdb/loqusdbapi misbehaving or have filter settings changed in MIP? On a quick look, it does seem we have some rather common variants "missing" from loqusdb, e.g. 

![Screenshot 2022-03-01 at 08 54 30](https://user-images.githubusercontent.com/758570/156128417-1b8da27c-1c54-4279-b577-f2d009935642.png)

Reproduces on stage:

```
$ curl -X GET "http://cg-vm1.scilifelab.se:7079/svs/?chrom=X&end_chrom=X&pos=65549568&end=88119178&sv_type=DEL"
Internal Server Error
```

```
Mar 01 09:27:41 cg-vm1.scilifelab.se conmon[340192]: 10.0.2.100:38494 - "GET /svs/?chrom=X&end_chrom=X&pos=65549568&end=88119178&sv_type=DEL HTTP/1.1" 500
Mar 01 09:27:41 cg-vm1.scilifelab.se podman[340135]: 10.0.2.100:38494 - "GET /svs/?chrom=X&end_chrom=X&pos=65549568&end=88119178&sv_type=DEL HTTP/1.1" 500
Mar 01 09:27:41 cg-vm1.scilifelab.se conmon[340192]: [2022-03-01 08:27:41 +0000] [4] [ERROR] Exception in ASGI application
Mar 01 09:27:41 cg-vm1.scilifelab.se conmon[340192]: Traceback (most recent call last):
Mar 01 09:27:41 cg-vm1.scilifelab.se conmon[340192]:   File "/usr/local/lib/python3.9/site-packages/uvicorn/protocols/http/httptools_impl.py", line 376, in run_asgi
Mar 01 09:27:41 cg-vm1.scilifelab.se conmon[340192]:     result = await app(self.scope, self.receive, self.send)
Mar 01 09:27:41 cg-vm1.scilifelab.se podman[340135]: [2022-03-01 08:27:41 +0000] [4] [ERROR] Exception in ASGI application
Mar 01 09:27:41 cg-vm1.scilifelab.se podman[340135]: Traceback (most recent call last):
Mar 01 09:27:41 cg-vm1.scilifelab.se podman[340135]:   File "/usr/local/lib/python3.9/site-packages/uvicorn/protocols/http/httptools_impl.py", line 376, in run_asgi
Mar 01 09:27:41 cg-vm1.scilifelab.se podman[340135]:     result = await app(self.scope, self.receive, self.send)
Mar 01 09:27:41 cg-vm1.scilifelab.se podman[340135]:   File "/usr/local/lib/python3.9/site-packages/uvicorn/middleware/proxy_headers.py", line 75, in __call__
Mar 01 09:27:41 cg-vm1.scilifelab.se podman[340135]:     return await self.app(scope, receive, send)
Mar 01 09:27:41 cg-vm1.scilifelab.se podman[340135]:   File "/usr/local/lib/python3.9/site-packages/fastapi/applications.py", line 179, in __call__
Mar 01 09:27:41 cg-vm1.scilifelab.se podman[340135]:     await super().__call__(scope, receive, send)
Mar 01 09:27:41 cg-vm1.scilifelab.se podman[340135]:   File "/usr/local/lib/python3.9/site-packages/starlette/applications.py", line 111, in __call__
Mar 01 09:27:41 cg-vm1.scilifelab.se podman[340135]:     await self.middleware_stack(scope, receive, send)
Mar 01 09:27:41 cg-vm1.scilifelab.se podman[340135]:   File "/usr/local/lib/python3.9/site-packages/starlette/middleware/errors.py", line 181, in __call__
Mar 01 09:27:41 cg-vm1.scilifelab.se podman[340135]:     raise exc from None
Mar 01 09:27:41 cg-vm1.scilifelab.se podman[340135]:   File "/usr/local/lib/python3.9/site-packages/starlette/middleware/errors.py", line 159, in __call__
Mar 01 09:27:41 cg-vm1.scilifelab.se podman[340135]:     await self.app(scope, receive, _send)
Mar 01 09:27:41 cg-vm1.scilifelab.se podman[340135]:   File "/usr/local/lib/python3.9/site-packages/starlette/exceptions.py", line 82, in __call__
Mar 01 09:27:41 cg-vm1.scilifelab.se podman[340135]:     raise exc from None
Mar 01 09:27:41 cg-vm1.scilifelab.se podman[340135]:   File "/usr/local/lib/python3.9/site-packages/starlette/exceptions.py", line 71, in __call__
Mar 01 09:27:41 cg-vm1.scilifelab.se podman[340135]:     await self.app(scope, receive, sender)
Mar 01 09:27:41 cg-vm1.scilifelab.se podman[340135]:   File "/usr/local/lib/python3.9/site-packages/starlette/routing.py", line 566, in __call__
Mar 01 09:27:41 cg-vm1.scilifelab.se podman[340135]:     await route.handle(scope, receive, send)
Mar 01 09:27:41 cg-vm1.scilifelab.se podman[340135]:   File "/usr/local/lib/python3.9/site-packages/starlette/routing.py", line 227, in handle
Mar 01 09:27:41 cg-vm1.scilifelab.se podman[340135]:     await self.app(scope, receive, send)
Mar 01 09:27:41 cg-vm1.scilifelab.se podman[340135]:   File "/usr/local/lib/python3.9/site-packages/starlette/routing.py", line 41, in app
Mar 01 09:27:41 cg-vm1.scilifelab.se podman[340135]:     response = await func(request)
Mar 01 09:27:41 cg-vm1.scilifelab.se podman[340135]:   File "/usr/local/lib/python3.9/site-packages/fastapi/routing.py", line 190, in app
Mar 01 09:27:41 cg-vm1.scilifelab.se podman[340135]:     response_data = await serialize_response(
Mar 01 09:27:41 cg-vm1.scilifelab.se podman[340135]:   File "/usr/local/lib/python3.9/site-packages/fastapi/routing.py", line 111, in serialize_response
Mar 01 09:27:41 cg-vm1.scilifelab.se podman[340135]:     raise ValidationError(errors, field.type_)
Mar 01 09:27:41 cg-vm1.scilifelab.se podman[340135]: pydantic.error_wrappers.ValidationError: 1 validation error for StructuralVariant
Mar 01 09:27:41 cg-vm1.scilifelab.se podman[340135]: response -> _id
Mar 01 09:27:41 cg-vm1.scilifelab.se conmon[340192]:   File "/usr/local/lib/python3.9/site-packages/uvicorn/middleware/proxy_headers.py", line 75, in __call__
Mar 01 09:27:41 cg-vm1.scilifelab.se podman[340135]:   str type expected (type=type_error.str)
Mar 01 09:27:41 cg-vm1.scilifelab.se conmon[340192]:     return await self.app(scope, receive, send)
Mar 01 09:27:41 cg-vm1.scilifelab.se conmon[340192]:   File "/usr/local/lib/python3.9/site-packages/fastapi/applications.py", line 179, in __call__
Mar 01 09:27:41 cg-vm1.scilifelab.se conmon[340192]:     await super().__call__(scope, receive, send)
Mar 01 09:27:41 cg-vm1.scilifelab.se conmon[340192]:   File "/usr/local/lib/python3.9/site-packages/starlette/applications.py", line 111, in __call__
Mar 01 09:27:41 cg-vm1.scilifelab.se conmon[340192]:     await self.middleware_stack(scope, receive, send)
Mar 01 09:27:41 cg-vm1.scilifelab.se conmon[340192]:   File "/usr/local/lib/python3.9/site-packages/starlette/middleware/errors.py", line 181, in __call__
Mar 01 09:27:41 cg-vm1.scilifelab.se conmon[340192]:     raise exc from None
Mar 01 09:27:41 cg-vm1.scilifelab.se conmon[340192]:   File "/usr/local/lib/python3.9/site-packages/starlette/middleware/errors.py", line 159, in __call__
Mar 01 09:27:41 cg-vm1.scilifelab.se conmon[340192]:     await self.app(scope, receive, _send)
Mar 01 09:27:41 cg-vm1.scilifelab.se conmon[340192]:   File "/usr/local/lib/python3.9/site-packages/starlette/exceptions.py", line 82, in __call__
Mar 01 09:27:41 cg-vm1.scilifelab.se conmon[340192]:     raise exc from None
Mar 01 09:27:41 cg-vm1.scilifelab.se conmon[340192]:   File "/usr/local/lib/python3.9/site-packages/starlette/exceptions.py", line 71, in __call__
Mar 01 09:27:41 cg-vm1.scilifelab.se conmon[340192]:     await self.app(scope, receive, sender)
Mar 01 09:27:41 cg-vm1.scilifelab.se conmon[340192]:   File "/usr/local/lib/python3.9/site-packages/starlette/routing.py", line 566, in __call__
Mar 01 09:27:41 cg-vm1.scilifelab.se conmon[340192]:     await route.handle(scope, receive, send)
Mar 01 09:27:41 cg-vm1.scilifelab.se conmon[340192]:   File "/usr/local/lib/python3.9/site-packages/starlette/routing.py", line 227, in handle
Mar 01 09:27:41 cg-vm1.scilifelab.se conmon[340192]:     await self.app(scope, receive, send)
Mar 01 09:27:41 cg-vm1.scilifelab.se conmon[340192]:   File "/usr/local/lib/python3.9/site-packages/starlette/routing.py", line 41, in app
Mar 01 09:27:41 cg-vm1.scilifelab.se conmon[340192]:     response = await func(request)
Mar 01 09:27:41 cg-vm1.scilifelab.se conmon[340192]:   File "/usr/local/lib/python3.9/site-packages/fastapi/routing.py", line 190, in app
Mar 01 09:27:41 cg-vm1.scilifelab.se conmon[340192]:     response_data = await serialize_response(
Mar 01 09:27:41 cg-vm1.scilifelab.se conmon[340192]:   File "/usr/local/lib/python3.9/site-packages/fastapi/routing.py", line 111, in serialize_response
Mar 01 09:27:41 cg-vm1.scilifelab.se conmon[340192]:     raise ValidationError(errors, field.type_)
Mar 01 09:27:41 cg-vm1.scilifelab.se conmon[340192]: pydantic.error_wrappers.ValidationError: 1 validation error for StructuralVariant
Mar 01 09:27:41 cg-vm1.scilifelab.se conmon[340192]: response -> _id
Mar 01 09:27:41 cg-vm1.scilifelab.se conmon[340192]:   str type expected (type=type_error.str)
```
But if we switch back to an old container:
```
[hiseq.clinical@cg-vm1 ~]$ systemctl --user start loqusdbapi-rd@vm-ready-package
[hiseq.clinical@cg-vm1 ~]$  curl -X GET "http://cg-vm1.scilifelab.se:7079/svs/?chrom=X&end_chrom=X&pos=65549568&end=88119178&sv_type=DEL"
{"chrom":"X","observations":1228,"families":["choiceracer","tightdoe","briefglider","primeasp","livingpup","topghost","tidygull","caringhound","neatkodiak","fluentsnake","trueroughy","movingbengal","superbhyena","growncrane","novelfilly","awareliger","frankswan","greatosprey","activehyena","mintlocust","suitedakita","subtlestag","poeticraptor","validmoose","intentboar","normallion","vocalrodent","gamegoblin","crackkodiak","moralmidge","actualfeline","squarequail","betterturtle","loyaloryx","onechimp","robustsnipe","happyrobin","busyloon","viableparrot","socialbream","truedeer","frankloon","neatmole","honestparrot","livingmarmot","freeelk","gladurchin","caringskunk","pumpedquail"],"total":5323,"end_chrom":"X","end_left":88116180,"end_right":88122180,"sv_type":"DEL","length":22569593,"pos_left":65546587,"pos_right":65552587}[hiseq.clinical@cg-vm1 ~]$ 
```

--- a/loqusdbapi/models.py
+++ b/loqusdbapi/models.py
@@ -3,6 +3,19 @@ from typing import Optional, List, Union, Any
 
 from pydantic import BaseModel, validator, Field
 
+from bson.objectid import ObjectId as BsonObjectId
+
+class PydanticObjectId(BsonObjectId):
+    @classmethod
+    def __get_validators__(cls):
+        yield cls.validate
+
+    @classmethod
+    def validate(cls, v):
+        if not isinstance(v, BsonObjectId):
+            raise TypeError('ObjectId required')
+        return str(v)
+
 class Individual(BaseModel):
     ind_id: str
     case_id: str
@@ -69,7 +82,7 @@ class Variant(BaseVariant):
 
 
 class StructuralVariant(BaseVariant):
-    id: Optional[Any] = Field(alias="_id")
+    id: Optional[BsonObjectId] = Field(alias="_id")
     end_chrom: str
     end_left: int
     end_right: int

--- a/loqusdbapi/models.py
+++ b/loqusdbapi/models.py
@@ -76,7 +76,7 @@ class Variant(BaseVariant):
 
 
 class StructuralVariant(BaseVariant):
-    id: BsonObjectId = Field(alias="_id")
+    id: Optional[PydanticObjectId] = Field(alias="_id")
     end_chrom: str
     end_left: int
     end_right: int

--- a/loqusdbapi/models.py
+++ b/loqusdbapi/models.py
@@ -65,7 +65,6 @@ class Variant(BaseVariant):
 
 
 class StructuralVariant(BaseVariant):
-    id: ObjectId = Field(alias="_id")
     end_chrom: str
     end_left: int
     end_right: int

--- a/loqusdbapi/models.py
+++ b/loqusdbapi/models.py
@@ -48,7 +48,7 @@ class Case(BaseModel):
 
 
 class BaseVariant(BaseModel):
-    id: str = Field(alias="_id")
+    id: Optional[Any] = Field(alias="_id")
     chrom: str
     observations: int
     families: List[str] = []
@@ -65,6 +65,7 @@ class Variant(BaseVariant):
 
 
 class StructuralVariant(BaseVariant):
+    id: ObjectId = Field(alias="_id")
     end_chrom: str
     end_left: int
     end_right: int

--- a/loqusdbapi/models.py
+++ b/loqusdbapi/models.py
@@ -48,13 +48,13 @@ class Case(BaseModel):
 
 
 class BaseVariant(BaseModel):
-    id: Optional[Any] = Field(alias="_id")
     chrom: str
     observations: int
     families: List[str] = []
 
 
 class Variant(BaseVariant):
+    id: str = Field(alias="_id")
     start: int
     end: int
     ref: str

--- a/loqusdbapi/models.py
+++ b/loqusdbapi/models.py
@@ -27,12 +27,6 @@ class Individual(BaseModel):
     profile: Optional[list] = []
     similar_samples: Optional[list] = []
 
-    @validator("id")
-    def id_to_str(cls, value):
-        if value:
-            return str(value)
-
-
 class Case(BaseModel):
     id: Optional[Any] = Field(alias="_id")
     case_id: str

--- a/loqusdbapi/models.py
+++ b/loqusdbapi/models.py
@@ -3,6 +3,7 @@ from typing import Optional, List, Union, Any
 
 from pydantic import BaseModel, validator, Field
 
+from bson
 
 class Individual(BaseModel):
     ind_id: str
@@ -14,6 +15,11 @@ class Individual(BaseModel):
     ind_index: Optional[int]
     profile: Optional[list] = []
     similar_samples: Optional[list] = []
+
+    @validator("id")
+    def id_to_str(cls, value):
+        if value:
+            return str(value)
 
 
 class Case(BaseModel):

--- a/loqusdbapi/models.py
+++ b/loqusdbapi/models.py
@@ -76,7 +76,7 @@ class Variant(BaseVariant):
 
 
 class StructuralVariant(BaseVariant):
-    id: Optional[BsonObjectId] = Field(alias="_id")
+    id: BsonObjectId = Field(alias="_id")
     end_chrom: str
     end_left: int
     end_right: int

--- a/loqusdbapi/models.py
+++ b/loqusdbapi/models.py
@@ -65,7 +65,7 @@ class Variant(BaseVariant):
 
 
 class StructuralVariant(BaseVariant):
-    id: str = Field(alias="_id")
+    id: Optional[Any] = Field(alias="_id")
     end_chrom: str
     end_left: int
     end_right: int
@@ -81,6 +81,10 @@ class StructuralVariant(BaseVariant):
     def id_to_str(cls, value):
         if value:
             return str(value)
+
+    class Config:
+        arbitrary_types_allowed = True
+        allow_population_by_field_name = True
 
 
 class Cases(BaseModel):

--- a/loqusdbapi/models.py
+++ b/loqusdbapi/models.py
@@ -88,6 +88,9 @@ class StructuralVariant(BaseVariant):
     pos_sum: int
     total: int
 
+    class Config:
+        arbitrary_types_allowed = True
+
 
 class Cases(BaseModel):
     nr_cases_snvs: Optional[int]

--- a/loqusdbapi/models.py
+++ b/loqusdbapi/models.py
@@ -3,18 +3,7 @@ from typing import Optional, List, Union, Any
 
 from pydantic import BaseModel, validator, Field
 
-from bson.objectid import ObjectId as BsonObjectId
-
-class PydanticObjectId(BsonObjectId):
-    @classmethod
-    def __get_validators__(cls):
-        yield cls.validate
-
-    @classmethod
-    def validate(cls, v):
-        if not isinstance(v, BsonObjectId):
-            raise TypeError('ObjectId required')
-        return str(v)
+from bson.objectid import ObjectId
 
 class Individual(BaseModel):
     ind_id: str
@@ -76,7 +65,7 @@ class Variant(BaseVariant):
 
 
 class StructuralVariant(BaseVariant):
-    id: Optional[PydanticObjectId] = Field(alias="_id")
+    id: str = Field(alias="_id")
     end_chrom: str
     end_left: int
     end_right: int
@@ -88,8 +77,10 @@ class StructuralVariant(BaseVariant):
     pos_sum: int
     total: int
 
-    class Config:
-        arbitrary_types_allowed = True
+    @validator("id")
+    def id_to_str(cls, value):
+        if value:
+            return str(value)
 
 
 class Cases(BaseModel):

--- a/loqusdbapi/models.py
+++ b/loqusdbapi/models.py
@@ -65,6 +65,7 @@ class Variant(BaseVariant):
 
 
 class StructuralVariant(BaseVariant):
+    id: Optional[Any] = Field(alias="_id")
     end_chrom: str
     end_left: int
     end_right: int

--- a/loqusdbapi/models.py
+++ b/loqusdbapi/models.py
@@ -3,8 +3,6 @@ from typing import Optional, List, Union, Any
 
 from pydantic import BaseModel, validator, Field
 
-from bson
-
 class Individual(BaseModel):
     ind_id: str
     case_id: str

--- a/tests/test_loqusdbapi.py
+++ b/tests/test_loqusdbapi.py
@@ -2,4 +2,4 @@ from loqusdbapi import __version__
 
 
 def test_version():
-    assert __version__ == '0.1.0'
+    assert __version__ == '0.1.5'

